### PR TITLE
Aggregated javadoc generation fix + missing keycloak-operator javadoc

### LINKED ISSUE #29816: Aggregated javadoc generation fix + missing keycloak-operator javadoc
### Before reporting an issue

- [X] I have read and understood the above terms for submitting issues, and I understand that my issue may be closed without action if I do not follow them.

### Area

dist/quarkus

### Describe the bug

Aggregated javadoc is not generated in the downstream build due to maven-javadoc-plugin misconfiguration.
+ Keycloak-operator javadoc is missing from the bundle of api-docs.

### Version

24.0.4

### Regression

- [ ] The issue is a regression

### Expected behavior

Downstream build has to produce proper keycloak-api-docs-dist-<version>.zip file. 
- containing both javadocs and rest-api content.

### Actual behavior

Content of javadocs is missing completely.

### How to Reproduce?

Run the downstream build (ping me for the details). 

### Anything else?

_No response_

### PR DESCRIPTION
**Pull request by pskopek**
Closes #29816

### CODE REVIEW

**Comment by mhajas** [comment]
FYI: I fixed the Javadoc generation issue here: https://github.com/keycloak/keycloak/pull/29815

**Comment by pskopek** [comment]
> FYI: I fixed the Javadoc generation issue here: #29815

@mhajas thanks for letting me know. This is a different issue though.

**Comment by mhajas** [comment]
Why I am commenting is because this PR is reintroducing the dependency on `owasp-java-html-sanitizer` to distribution/api-docs-dist/pom.xml which was the root cause of the #29814 so it seemed there is some connection between these two.

**Comment by ahus1** [comment]
@pskopek - fyi, this PR now acquired a merge conflict in the `distribution/api-docs-dist/pom.xml`

**Comment by pskopek** [comment]
> @pskopek - fyi, this PR now acquired a merge conflict in the `distribution/api-docs-dist/pom.xml`

Thanks @ahus1, rebased.

File: distribution/api-docs-dist/pom.xml [hunk 1]
```diff
@@ -34,6 +34,24 @@
         <maven.compiler.release>17</maven.compiler.release>
     </properties>
 
+    <dependencyManagement>
+        <dependencies>
+            <dependency>
+                <groupId>com.redhat.quarkus.platform</groupId>
```
Comment (ahus1) [inline distribution/api-docs-dist/pom.xml:40]: The `com.redhat` group ID is not available upstream. To be able to build it in upstream, I changed it to `io.quarkus`. Would that still work in downstream?

```suggestion
                <groupId>io.quarkus.platform</groupId>
```

**Review by ahus1 — CHANGES_REQUESTED** [review]
Thank you for the PR. I found that it doesn't build upstream, see below for a suggestion.

File: distribution/api-docs-dist/pom.xml (hunk 1 above)
Comment (pskopek) [reply distribution/api-docs-dist/pom.xml:40]: Yes, you  are right. I am sorry for it.

**Review by ahus1 — APPROVED** [review]

**Pull request merged**